k2/csrc/fsa_utils_test.cu

```diff
@@ -651,4 +655,55 @@ TEST_F(StatesBatchSuiteTest, TestBackwardScores) {
   // host code?)
 }
 
+TEST_F(StatesBatchSuiteTest, TestArcScores) {
+  ContextPtr cpu = GetCpuContext();  // will use to copy data
+  {
+    // simple case
+    for (auto &context : {GetCpuContext(), GetCudaContext()}) {
+      FsaVec fsa_vec = fsa_vec_.To(context);
+      int32_t num_fsas = fsa_vec.Dim0(), num_states = fsa_vec.TotSize(1),
+              num_arcs = fsa_vec.NumElements();
+      EXPECT_EQ(num_fsas, 3);
+
+      Ragged<int32_t> state_batches = GetStateBatches(fsa_vec, true);
+      Array1<int32_t> dest_states = GetDestStates(fsa_vec, true);
+      Ragged<int32_t> incoming_arcs = GetIncomingArcs(fsa_vec, dest_states);
+      Ragged<int32_t> entering_arc_batches =
+          GetEnteringArcIndexBatches(fsa_vec, incoming_arcs, state_batches);
+      Ragged<int32_t> leaving_arc_batches =
+          GetLeavingArcIndexBatches(fsa_vec, state_batches);
+
+      {
+        // max
+        Array1<float> forward_scores = GetForwardScores<float>(
+            fsa_vec, state_batches, entering_arc_batches, false);
+        Array1<float> backward_scores = GetBackwardScores<float>(
+            fsa_vec, state_batches, leaving_arc_batches, nullptr, false);
+        Array1<float> arc_scores =
+            GetArcScores(fsa_vec, forward_scores, backward_scores);
+        EXPECT_EQ(arc_scores.Dim(), num_arcs);
+        K2_LOG(INFO) << arc_scores;
```

Comment:
I would propose to use
https://github.com/k2-fsa/k2/blob/db42b02b3672c39d2b1de6229d8ce23e774fb1cd/k2/csrc/array.h#L257

to get the values of a given array of indexes and check them.

Reply:
mm, actually I have not checked the values currently, I plan to compare it with host code result automatically, so please ignore this for now.
https://github.com/k2-fsa/k2/blob/db42b02b3672c39d2b1de6229d8ce23e774fb1cd/k2/csrc/fsa_utils_test.cu#L650-L651